tools/doc/allhtml.mjs

```diff
@@ -31,13 +31,35 @@ for (const link of toc.match(/<a.*?>/g)) {
   // Split the doc.
   const match = /(<\/ul>\s*)?<\/\w+>\s*<\w+ id="apicontent">/.exec(data);
 
+  // Get module name
+  const moduleName = href.split('.')[0];
+
   contents += data.slice(0, match.index)
-    .replace(/[\s\S]*?id="toc"[^>]*>\s*<\w+>.*?<\/\w+>\s*(<ul>\s*)?/, '');
+    .replace(/[\s\S]*?id="toc"[^>]*>\s*<\w+>.*?<\/\w+>\s*(<ul>\s*)?/, '')
+    // Prefix TOC links with current module name
+    .replace(/<a href="#([^"]+)"/g, (match, anchor) => {
+      return `<a href="#${moduleName}_${anchor}"`;
+    });
 
   apicontent += '<section>' + data.slice(match.index + match[0].length)
     .replace(/<!-- API END -->[\s\S]*/, '</section>')
+    // Prefix all in-page id's with module name (mostly special cases)
+    .replace(/id="([^"]+)"/g, (match, anchor) => {
+      return `id="${moduleName}_${anchor}"`;
+    })
+    // Prefix all in-page anchor marks with module name
+    .replace(/<a class="mark" href="#([^"]+)" id="([^"]+)"/g, (match, anchor) => {
+      return `<a class="mark" href="#${moduleName}_${anchor}" id="${moduleName}_${anchor}"`;
```

Comment:
Shouldn't this be
```suggestion
    .replace(/<a class="mark" href="#([^"]+)" id="([^"]+)"/g, (match, anchor, id) => {
      return `<a class="mark" href="#${moduleName}_${anchor}" id="${moduleName}_${id}"`;
```
or maybe
```suggestion
    .replace(/<a class="mark" href="#([^"]+)" id="([^"]+)"/g, (match, anchor, id) => {
      return anchor === id ?
        `<a class="mark" href="#${moduleName}_${anchor}" id="${moduleName}_${anchor}"`:
        match; // if anchor and id are different, do not change anything
```

Otherwise, stuff like https://nodejs.org/docs/latest/api/modules.html#modules_module_builtinmodules would break.

Reply:
I should have fixed this in 778c162. Those seemed to be the only hardcoded links that also acted as aliases. Is that a sufficient resolution?

Reply:
Sure thing - added in 9eb0cdd.